### PR TITLE
Ensure component updates are executed at the correct recyclerView state

### PR DESCRIPTION
**Pull request by fkose696**
fixes https://github.com/Comcast/blueprint/issues/50

### CODE REVIEW

File: library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.SCROLL_STATE_IDLE
+import com.xfinity.blueprint.model.Component
+
+class ViewAdapter(private val recyclerView: RecyclerView,
+                  private var loadingView: View? = null) : ViewRegistry {
+
+    override fun updateComponents(newComponents: List<Component>, behavior: () -> Unit) {
```
Comment (mrtvrgn) [inline library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt]: You don't seem to update anything. newComponents param is in no use too.

File: library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        if (!recyclerView.isComputingLayout) {
+            behavior.invoke()
+            return
+        }
+        recyclerView.addOnScrollListener(RecyclerViewScrollHandler(behavior))
```
Comment (mrtvrgn) [inline library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt]: Doesn't this add a new listener every time we update components?

File: library/src/main/java/com/xfinity/blueprint/view/ScreenViewDelegate.kt [hunk 3]
```diff
@@ -92,7 +92,9 @@ open class ScreenViewDelegate(componentRegistry: ComponentRegistry,
      * @param newComponents A List of the components the screen should be updated to display
      */
     override fun updateComponents(newComponents: List<Component>) {
-        updateComponents(newComponents, { oldComponents -> DefaultComponentDiffCallback(oldComponents, newComponents) })
+        viewRegistry.updateComponents(newComponents) {
```
Comment (mrtvrgn) [inline library/src/main/java/com/xfinity/blueprint/view/ScreenViewDelegate.kt]: I feel the chained calls like this are problematic. In the view registry, you are not actually updating anything.

File: library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    override fun showLoading() {
+        loadingView?.visibility = View.VISIBLE
+    }
+
+    override fun hideLoading() {
+        loadingView?.visibility = View.GONE
+    }
```
Comment (mrtvrgn) [inline library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt]: What's the benefit of having moving loading work here? Is screenviewdelegate delegating also delegating?

File: library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt (hunk 1 above)
Comment (fkose696) [reply library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt]: Updates are happening. I agree that newComponents param looks redundant in this case.

File: library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt (hunk 2 above)
Comment (fkose696) [reply library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt]: Yes, eventually listener will need to be set to null

File: library/src/main/java/com/xfinity/blueprint/view/ScreenViewDelegate.kt (hunk 3 above)
Comment (fkose696) [reply library/src/main/java/com/xfinity/blueprint/view/ScreenViewDelegate.kt]: Updates are happening. Can you clarify?

File: library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt (hunk 4 above)
Comment (fkose696) [reply library/src/main/java/com/xfinity/blueprint/ViewAdapter.kt]: To abstract away View from the Delegate.

**Comment by dapp** [comment]
replaced by https://github.com/Comcast/blueprint/pull/54

**Pull request closed without merging**